### ISSUE TITLE
tensorflow and skimage.io.imread issue with PIL plugin

### ISSUE THREAD

**Issue by liuyifly06** [scikit-image/scikit-image#2000]
When I was running following code
```
import tensorflow as tf
import sys, traceback, time
import numpy as np
from matplotlib import pyplot as plt
from skimage import io
def generateInstancesNN(instanceSize, step, plot_show = 1):
    ...
    train_file = "../../images/detector_5_no_5_angle_2.jpg"
    train = io.imread(train_file)
    ...
```
A error appeared:

```
Traceback (most recent call last):
  File "trainDataSet.py", line 312, in main
    read_data_sets(instanceSize, step, plot_show = 1)
  File "trainDataSet.py", line 67, in read_data_sets
    [train_images, train_labels] = generateInstancesNN(instanceSize, step, plot_show)
  File "trainDataSet.py", line 76, in generateInstancesNN
   ** train = io.imread(train_file)**
  File "/usr/local/lib/python2.7/dist-packages/skimage/io/_io.py", line 61, in imread
    img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
  File "/usr/local/lib/python2.7/dist-packages/skimage/io/manage_plugins.py", line 211, in call_plugin
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/skimage/io/_plugins/pil_plugin.py", line 37, in imread
    return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
  File "/usr/local/lib/python2.7/dist-packages/skimage/io/_plugins/pil_plugin.py", line 61, in pil_to_ndarray
    raise ValueError(error_message)
ValueError: Could not load "" 
**Reason: "broken data stream when reading image file"**
Please see documentation at: http://pillow.readthedocs.org/en/latest/installation.html#external-libraries
```
I reinstalled pillow and everything, this error just appears ...

However, the weird thing is that if I just type the code in python as following, everything works !

```
$python
>>>from skimage import io
>>>data = io.imread("../../images/detector_5_no_5_angle_2.jpg");
>>>data.shape
(2160, 4096, 3)
```
Why ??????? 
Why do I have this error just in running this file ???
I type the code into python, then it works. help me !

**Comment by emmanuelle** [comment]
Hi @liuyifly06 this is weird indeed. How did you execute your script? With "python trainDataSet.py" in a terminal? Within IPython? 

I would suggest that you try to use the debugger within IPython, as explained here http://www.scipy-lectures.org/advanced/debugging/index.html#postmortem
In the debugger, you can navigate between function calls using "u" (up) and "d" (down). 

If you go inside ``generateInstancesNN``, you could try to open the image with other functions, such as ``matplotlib.pyplot.imread``, and get more information to understand where the issue comes from.

**Comment by liuyifly06** [comment]
@emmanuelle thank you for your help. I am running with "python trainDataSet.py" in a terminal. I will try to debug it with IPython. 

Actually the program is running perfect on another Linux machine, which makes me wondering what's wrong with my computer. This kind of thing really sucks.

**Comment by soupault** [comment]
@liuyifly06 Does `read_data_sets` contain any parallel code? Probably, two concurrent processes are trying to access the same file.

**Comment by soupault** [comment]
@liuyifly06 Any news on the topic?

**Comment by liuyifly06** [comment]
@soupault I don not think I have any parallel code. I have uninstall and reinstall everything for many times.  Currently, for catching up my project deadline, I am just using another Linux machine on which the code can successfully run. I am going to check the problem again in next week. I will let you know if I have any progress.  

Very appreciate for your help!

**Comment by soupault** [comment]
@liuyifly06 It would be perfect if you could create minimal example which you'll be able to fully share. That would really facilitate the debugging process.

**Comment by liuyifly06** [comment]
@soupault Thanks, I will organize and upload my code and all the error information in a few days.

**Comment by soupault** [comment]
Ping @liuyifly06

**Comment by liuyifly06** [comment]
@soupault  working on it. Just finished my spring break vacation

**Comment by liuyifly06** [comment]
@soupault 

Still working on it.  

At current stage, I have identified that the problem only appears when I import `tensorflow`

For example, when I run following code: 

```
import tensorflow as tf
import sys, traceback, time
from skimage import io

def generateInstancesNN(instanceSize, step, label_option, plot_show = 1):
    train_file = "../../images/detector_5_no_5_angle_2.jpg"
    train = io.imread(train_file)
    return train

def main():
     try:
         instanceSize = 20;
         step = 10;
         edge = 4;
         scale =10;
	 label_option = 2;
	 generateInstancesNN(instanceSize, step, label_option, plot_show = 1)
         #read_data_sets(instanceSize, step, label_option, plot_show = 1)
	 
     except KeyboardInterrupt:
         print "Shutdown requested... exiting"
     except Exception:
         traceback.print_exc(file=sys.stdout)
     sys.exit(0)
 
if __name__ == '__main__':
     main()
```

Following error information appears:

```
Traceback (most recent call last):
  File "debug.py", line 17, in main
    generateInstancesNN(instanceSize, step, label_option, plot_show = 1)
  File "debug.py", line 7, in generateInstancesNN
    train = io.imread(train_file)
  File "/usr/local/lib/python2.7/dist-packages/scikit_image-0.12.3-py2.7-linux-x86_64.egg/skimage/io/_io.py", line 61, in imread
    img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
  File "/usr/local/lib/python2.7/dist-packages/scikit_image-0.12.3-py2.7-linux-x86_64.egg/skimage/io/manage_plugins.py", line 211, in call_plugin
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/scikit_image-0.12.3-py2.7-linux-x86_64.egg/skimage/io/_plugins/pil_plugin.py", line 37, in imread
    return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
  File "/usr/local/lib/python2.7/dist-packages/scikit_image-0.12.3-py2.7-linux-x86_64.egg/skimage/io/_plugins/pil_plugin.py", line 61, in pil_to_ndarray
    raise ValueError(error_message)
ValueError: Could not load "" 
Reason: "broken data stream when reading image file"
Please see documentation at: http://pillow.readthedocs.org/en/latest/installation.html#external-libraries
```

**If you eliminate the first line of the source code** (`import tensorflow as tf`), **the code can successfully run.**

I installed GPU enabled [TensorFlow](https://www.tensorflow.org/).

Trying to figure out why GPU enabled TensorFlow result in this problem on my computer.

**Comment by liuyifly06** [comment]
Updated last comment.

**Comment by soupault** [comment]
Can't reproduce with `tensorflow 0.7.1 CPU`, `skimage 0.12.3`, `Python 2.7.11`.
Unfortunately, I don't have platform where to install `tensorflow GPU`.

**Comment by Seth-Park** [comment]
I tried the code in my platform (NVIDIA K80 GPU supported, tensorflow 0.7.1 GPU, skimage 0.12.3, Python 2.7.11 Anaconda) and it works fine. I first thought that the jpg file might have been corrupted, but since changing the order of imports solves the problem, I am wondering what could have gone wrong...

**Comment by stefanv** [comment]
Thanks, Seth, that's helpful.

**Comment by stefanv** [comment]
This doesn't seem to be an skimage issue, so I'm closing for now, but feel free to re-open if there's any proof to the contrary.

**Comment by eriophora** [comment]
FYI, I'm getting this same issue when trying to modify tensorflow serving. It works perfectly fine in the interpreter, but when it's an executable from `bazel build` PIL freaks out.

Further, I don't think it's a problem with multiple open files--even if it's read into memory and fed to PIL as a StringIO then shunted directly into a numpy array all under a concurrency lock it still freaks out with the same error.

**Comment by sehugg** [comment]
Same here, when I use imread after importing Tensorflow (even indirectly) I get a PIL.JpegImagePlugin.JpegImageFile object returned instead of ndarray.

**Comment by soupault** [comment]
@sehugg Could you report this to https://github.com/python-pillow/Pillow ?

**Comment by liuyifly06** [comment]
@soupault @sehugg @eriophora 
I submitted the issue to TensorFlow. It turns out the developer of TensorFLow already acknowledge this problem.
[https://github.com/tensorflow/tensorflow/issues/2101](https://github.com/tensorflow/tensorflow/issues/2101)
[https://github.com/tensorflow/tensorflow/issues/1924](https://github.com/tensorflow/tensorflow/issues/1924)
I guess you could expect improving of TensorFlow in next version.

Basically two solutions right now:
1. import skimage before TensorFlow 
2. I know TensorFlow 0.7.1 version works well without any conflicts. I am using that version right now.

**Comment by hmaarrfk** [comment]
The previous issue is from 2016 and likely has very old information.

To allow us to help you best, please open a new issue. We ask for a little bit of information that helps us situate you best.
